src/player/watch_media.rs

```diff
@@ -11,30 +11,49 @@ use crate::{
     media::Media,
     player::mpv::open_mpv,
     player::vlc::open_vlc,
-    TRANSLATION, USE_MPV,
+    TRANSLATION, USE_GECKODRIVER, USE_MPV,
 };
 
 #[tokio::main]
 pub async fn watch_media(media: Media, img_mode: Option<bool>) -> WebDriverResult<()> {
     let language = TRANSLATION.get().unwrap();
     let use_mpv = USE_MPV.get().unwrap();
+    let use_geckodriver = USE_GECKODRIVER.get().unwrap();
 
-    let url = format!("https://vizer.in/{}", &media.url);
-    let mut chromedriver = Command::new("chromedriver").spawn().unwrap();
-    // we need to wait chromedriver to start :(
+    let mut command_name = String::from("chromedriver");
+    if *use_geckodriver {
+        command_name.clear();
+        command_name.push_str("geckodriver");
+    };
+
+    let mut command = Command::new(command_name)
+        .stdout(Stdio::null())
+        .stderr(Stdio::null())
+        .spawn()
+        .unwrap();
+    // we need to wait command to start :(
     sleep(Duration::from_millis(100));
 
+    let url = format!("https://vizer.in/{}", &media.url);
+
     print!("\x1B[2J\x1B[1;1H");
     println!("{}", language.preparing_misc_text);
 
-    let mut caps = DesiredCapabilities::chrome();
-    caps.set_headless().unwrap();
-    let driver = WebDriver::new("http://localhost:9515", caps).await?;
+    let mut driver = Vec::new();
```

Comment:
may have a better way to do this.

Reply:
I think a better approach would be creating a struct WebDriver with the fields `browser`, `port` and `capabilities` and pass that instance as a parameter of the function watch_media

Reply:
i see, working on it :), but i find something, the capabilities changes of type when calling DesiredCapabilities::<Some Browser>, so i can't parse the direct capabilities via one struct as the type in one struct field can't be changed.

DesiredCapabilities::firefox(); = FirefoxCapabilities;
DesiredCapabilities::chrome(); = ChromeCapabilities;

Reply:
have method that removes the vector, but with this method the driver is inicializated with chrome capabilities and if the flag -g is parsed it will quit the server with chrome capabilities and run another server with the firefox capabilities,

 the problem with this method is that if the -g is passed the server will still be created with chrome capabilities and quit without any use for it
 
 ```
   let mut caps = DesiredCapabilities::chrome();
    caps.set_headless().unwrap();
    let mut driver = WebDriver::new("9595", caps).await?;

        if *use_geckodriver {
            driver.quit().await?;
            let mut caps = DesiredCapabilities::firefox();
            caps.set_headless().unwrap();
            driver = WebDriver::new("4444", caps).await?;
        }
```

Reply:
if wanted i can pass the port data, and the browser flag via one struct like you said, but i don't know how to parse the capabilities :/.

Reply:
Well you can use enums like this to parse the capabilities:
```rust
pub struct WebDriverss {
    pub browser: Browser,
}

pub enum Browser {
    Chrome(ChromeCapabilities),
    Firefox(FirefoxCapabilities),
}

    let webdriver: WebDriverss = WebDriverss {
        browser: Browser::Chrome(DesiredCapabilities::chrome()),
    };

    match webdriver.browser {
        Browser::Chrome(a) => {
            let driver = WebDriver::new("http://localhost:9515", a).await?;
        }
        Browser::Firefox(a) => {
            let driver = WebDriver::new("http://localhost:9515", a).await?;
        }
        _ => print!(""),
    }

```

Reply:
but maybe using just an if statement to define the capabilities is better, bc I don't plan to use other browsers

Reply:
i was looking to another way to use the driver without the vector being used to hold it value but i can't think in one, the enum won't work for it

```    
    match webdriver.browser {
        Browser::Chrome(a) => {
            let driver = WebDriver::new("http://localhost:9515", a).await?;
        }
        Browser::Firefox(a) => {
            let driver = WebDriver::new("http://localhost:9515", a).await?; 
        }                    
        _ => print!(""),
    }
                       
        | - i can't acess the driver value here.       
        V                                                  
    driver.do_something(); 
```
- this works to parse the capabilities but the driver value is locked inside the match curly brackets
-   is the same thing that this code below

```
    if *use_geckodriver {
        let mut caps = DesiredCapabilities::firefox();
        caps.set_headless().unwrap();
        let driver = (WebDriver::new("http://localhost:4444", caps).await?);
    } else {
        let mut caps = DesiredCapabilities::chrome();
        caps.set_headless().unwrap();
        let driver = (WebDriver::new("http://localhost:9515", caps).await?);
    };
                                                                                                ^

   | it won't work because the driver only exist inside the above curly brackets
   v
driver.do_something()
```    

that's why i was using one vector to push the driver into it. using the enum would make easier to implement new browsers in the future but it won't remove the vector :(, 

and as you said that you don't plan to implement new browsers what method should i use, enum or if?

Reply:
Use the if statement

Reply:
And you can use something like this:
```rust
    let driver: WebDriver;

    if *use_geckodriver {
        let mut caps = DesiredCapabilities::firefox();
        caps.set_headless().unwrap();
        driver = WebDriver::new("http://localhost:4444", caps).await?;
    } else {
        let mut caps = DesiredCapabilities::chrome();
        caps.set_headless().unwrap();
        driver = WebDriver::new("http://localhost:9515", caps).await?;
    };

    driver.do_something()
```